modules/couchbase/src/main/java/org/testcontainers/containers/CouchbaseContainer.java

```diff
@@ -0,0 +1,284 @@
+        this.index = withIndex;
+        return self();
+    }
+
+    public SELF withQuery(Boolean withQuery) {
+        this.query = withQuery;
+        return self();
+    }
+
+    public SELF withFTS(Boolean withFTS) {
+        this.fts = withFTS;
+        return self();
+    }
+
+    public SELF withTravelSample(Boolean withTravelSample) {
+        this.travelSample = withTravelSample;
+        return self();
+    }
+
+    public SELF withBeerSample(Boolean withBeerSample) {
+        this.beerSample = withBeerSample;
+        return self();
+    }
+
+    public SELF withGamesIMSample(Boolean withGamesIMSample) {
+        this.gamesIMSample = withGamesIMSample;
+        return self();
+    }
+
+    public SELF withNewBucket(BucketSettings bucketSettings) {
+        newBuckets.add(bucketSettings);
+        return self();
+    }
+
+
+    public void initCluster() {
+        urlBase = String.format("http://%s:%s", getContainerIpAddress(), getMappedPort(8091));
+        try {
+            String poolURL = "/pools/default";
+            String poolPayload = "memoryQuota=" + URLEncoder.encode(memoryQuota, "UTF-8") + "&indexMemoryQuota=" + URLEncoder.encode(indexMemoryQuota, "UTF-8");
+
+            String setupServicesURL = "/node/controller/setupServices";
+            StringBuilder servicePayloadBuilder = new StringBuilder();
+            if (keyValue) {
+                servicePayloadBuilder.append("kv,");
+            }
+            if (query) {
+                servicePayloadBuilder.append("n1ql,");
+            }
+            if (index) {
+                servicePayloadBuilder.append("index,");
+            }
+            if (fts) {
+                servicePayloadBuilder.append("fts,");
+            }
+            String setupServiceContent = "services=" + URLEncoder.encode(servicePayloadBuilder.toString(), "UTF-8");
+
+            String webSettingsURL = "/settings/web";
+            String webSettingsContent = "username=" + URLEncoder.encode(clusterUsername, "UTF-8") + "&password=" + URLEncoder.encode(clusterPassword, "UTF-8") + "&port=8091";
+
+            String bucketURL = "/sampleBuckets/install";
+
+            StringBuilder sampleBucketPayloadBuilder = new StringBuilder();
+            sampleBucketPayloadBuilder.append('[');
+            if (travelSample) {
+                sampleBucketPayloadBuilder.append("\"travel-sample\",");
+            }
+            if (beerSample) {
+                sampleBucketPayloadBuilder.append("\"beer-sample\",");
+            }
+            if (gamesIMSample) {
+                sampleBucketPayloadBuilder.append("\"gamesim-sample\",");
+            }
+            sampleBucketPayloadBuilder.append(']');
+
+            callCouchbaseRestAPI(poolURL, poolPayload);
+            callCouchbaseRestAPI(setupServicesURL, setupServiceContent);
+            callCouchbaseRestAPI(webSettingsURL, webSettingsContent);
+            callCouchbaseRestAPI(bucketURL, sampleBucketPayloadBuilder.toString());
```

Comment:
TestContainers has a lifecycle, you can put this code to `containerIsStarting` (means that container was created) method, and then, in `containerIsStarted` (container passed the wait strategy), do
```
callCouchbaseRestAPI("/settings/indexes", "indexerThreads=0&logLevel=info&maxRollbackPoints=5&storageMode=memory_optimized");
```